utilities/console.py

```diff
@@ -40,6 +42,7 @@ def __init__(self, vm, username=None, password=None, timeout=30, prompt=None):
         self.cmd = self._generate_cmd()
         self.base_dir = get_data_collector_base_directory()
 
+    @retry(wait_timeout=TIMEOUT_10MIN, sleep=TIMEOUT_10SEC)
```

Comment:
I am pretty sure a 10min timeout will be nacked by the root maintainers.
How did you reach this number?

Reply:
I took the time we wait inside _connect and added a bit to is so that at lease a second retry will be triggered if _connect fails on timeout. Currently it has 5+2.5 min, so I gave it a 33% addition.
I thought the we might want to change the time we wait inside _connect(), it seems a lot on it's own  but this is not in the scope of the current PR.
Currently _connect() has a 5 min timeout waiting for the login prompt, (which I think can be reduced to 3-3.5 min, but we need to check this). and than another 2.5 min timeout waiting for the prompt (which also seems a lot).
If we reduce these times (this will require extensive tests to make sure the new timeout are enough), the 10 min can be narrowed.

Reply:
Thanks.

It will be useful to add the mentioned logic and reasoning to the commit message (and if missing, to the PR description). Future us may want to know what happened here and why.

After that, you can resolve this thread.